lib/abstractPlugin.js

```diff
@@ -0,0 +1,13 @@
+'use strict'
+
+const { kSetInstance } = require('./symbols.js')
+
+module.exports = class AbstractPlugin {
+  constructor () {
+    this.instance = null
+  }
+
+  [kSetInstance] (instance) {
+    this.instance = instance
+  }
```

Comment:
I don’t think this class is needed at all. We can just set the instance manually on the object, without using an accessor.